programs/merkle-distributor/src/lib.rs

```diff
@@ -22,6 +22,16 @@ pub mod merkle_proof;
 
 declare_id!("PMRKTWvK9f1cPkQuXvvyDPmyCSoq8FdedCimXrXJp8M");
 
+// Load an account that may be empty. If it's empty return None.
+fn load_optional_account<'a, T: AccountSerialize + AccountDeserialize + Owner + Clone>(
+    info: &AccountInfo<'a>,
+) -> Result<Option<Account<'a, T>>> {
+    if **info.lamports.borrow() == 0u64 {
```

Comment:
could there be a DOS attack, where i send 0.001 SOL to the expected address, and this function would always fail?

maybe it's better to check if the account data has 0 byte?